Add device-restricted off to min brightness feature

EPMatt opened the pull request:
In IKEA Zigbee implementation controllers have the ability to turn on the connected bulb (or light group) at minimum brightness if the action to increment brightness occurs (i.e. pressing the dedicated button for a `E1810`, turning the device to the right for a `ICTC-G-1` etc.).

This PR implements the feature, which can be device restricted, since not all controllers which will be integrated in the future could be designed to have this capability.
The `supports_from_off_to_min_brightness()` method, which can be overridden by every controller, defines whether the device supports this feature or not. By default, devices will not implement this feature.

Also, I had to edit `before_action()` signatures, passing action arguments to the methods, something that could come in handy if more complex checks will have to be performed in the future.

I succesfully tested and enabled the feature on `E1810` and `E1743` controllers. I left the `ICTC-G-1` unsupported since the integration is quite unstable and there are still issues with handling duplicate and unwanted messages.

xaviml commented on apps/z2m_ikea_controller/z2m_ikea_controller.py:
```diff
@@ -17,10 +17,10 @@
 
 
 def action(method):
-    def _action_impl(self, *args, **kwargs):
```
We currently not use any keyword arguments, but is always a good practice to leave them as well as adding `*args`. So I would suggest to not delete it.

EPMatt replied:
Ok, I'm adding `**kwargs` back in the next commit

xaviml commented on apps/z2m_ikea_controller/z2m_ikea_controller.py:
```diff
@@ -17,10 +17,10 @@
-    def _action_impl(self, *args, **kwargs):
-        continue_call = self.before_action(method.__name__)
+    def _action_impl(self, *args):
+        continue_call = self.before_action(method.__name__, *args)
```
Same would go for here, I would suggest to add the keyword arguments
```suggestion
continue_call = self.before_action(method.__name__, *args, **kwargs)
```

xaviml commented on apps/z2m_ikea_controller/z2m_ikea_controller.py:
```diff
@@ -17,10 +17,10 @@
+    def _action_impl(self, *args):
+        continue_call = self.before_action(method.__name__, *args)
         if continue_call:
-            method(self, *args, **kwargs)
```
Same as before

xaviml commented on apps/z2m_ikea_controller/z2m_ikea_controller.py:
```diff
@@ -320,6 +322,8 @@ def change_light_state(self, old, attribute, direction, steps):
         min_ = self.attribute_minmax[attribute]["min"]
         step = (max_ - min_) // steps
         new_state_attribute = old + sign * step
+        if self.from_off_to_min_brightness and attribute == "brightness" and  self.get_state(self.light["name"])=="off":
```
```suggestion
if self.from_off_to_min_brightness and attribute == ATTRIBUTE_BRIGHTNESS and  self.get_state(self.light["name"])=="off":
```
Also, I would put this logic on the `click` function since this function is called as well for `hold` as well. Or should the light also turn on when the brightness up is hold?

EPMatt replied:
My mistake, I didn't notice the `ATTRIBUTE_BRIGHTNESS` global.

EPMatt replied:
Yeah I think that also just clicking once the brightness up button on a controller would turn the lamp on at minimum brightness, but I'm not sure. Tomorrow I'll perform some tests with an `E1810` and a random IKEA bulb then I'll report the results back here.

EPMatt commented:
Hi @xaviml,
I made some tests today with a `E1810` and an IKEA bulb paired together and I confirm that, both with holding and clicking the `brightness_up` button when the light is off, the lamp is turned on at minimum brightness.
I updated the branch with the suggested changes, enabling the user to specify whether to enable this feature or not for the single app instance, and removed the defaults for `E1810` and `E1743` controllers.
If everything is fine and the PR is merged we should also update docs including the new feature.

xaviml commented:
@EPMatt Thanks you for your contribution! This will be included in the next release.

EPMatt commented:
You're welcome @xaviml! 👍
